string data
---
int64 key